akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolConductor.scala

```diff
@@ -63,17 +67,18 @@ private object PoolConductor {
                                   +---------+
 
   */
-  def apply(slotCount: Int, pipeliningLimit: Int, log: LoggingAdapter): Graph[Ports, Any] =
+  def apply(slotSettings: PoolSlotsSetting, pipeliningLimit: Int, log: LoggingAdapter): Graph[Ports, Any] =
```

Comment:
Without the extra class the function would look like this `apply(Int, Int, Int, LoggingAdapter)`. IMO Those 3 `Int`s make it easy to make a mistake. A mistake that won't be caught on compile time.
When it comes to naming - I feel like `Range` suggests that it holds some data (like the built-in one). Where in this case it's more like it hold two extremes, no the data in-between. But I'm not going to fight over it - if you prefer `Range` I will rename.